app/controllers/api/v1/profiles_controller.rb

```diff
@@ -1,9 +1,22 @@
 class Api::V1::ProfilesController < Api::BaseController
+
   def show
     @user = User.find_by_slug!(params[:id])
     respond_to do |format|
       format.json { render json: @user }
       format.yaml { render yaml: @user }
     end
   end
+
+  def me
+    authenticate_or_request_with_http_basic do |username, password|
+      user = AuthenticatedUser.authenticate(username.strip, password)
+      if user
```

Comment:
:thinking: Those two lines could be merged together (if Rubocop will not be against this) to:

```ruby
if user = AuthenticatedUser.authenticate(username.strip, password)
```

If user auth fails (and `authenticate` method returns `nil`), what is going to be rendered? Isn't there explicit render needed like in api_keys controller? :thinking: 

https://github.com/rubygems/rubygems.org/blob/1b75f7a8d059042d8f3faa4075f09d3b3988d96e/app/controllers/api/v1/api_keys_controller.rb#L54

Reply:
If authentication fails a 401 unauthorized is returned. [This test](https://github.com/rubygems/rubygems.org/pull/2922/files#diff-46e9d675e9328756d8b191629ad8c169f522a6a4d90e3f4ac299839458f2bd47R78) validates that for us. 

We get that protection by wrapping the render in the `if AuthenticatedUser.authenticate`. Without the if it would call our render regardless of whether authentication fails. With the `if` it falls back to the return value of `authenticate_or_request_with_http_basic` when authentication fails, which gives us the 401 unauthorized

Reply:
Thanks for the explanation. This works well, but I think we can do better by respecting the format and provide proper error message. Something similar is already happening at dependencies API controller.

https://github.com/rubygems/rubygems.org/blob/1b75f7a8d059042d8f3faa4075f09d3b3988d96e/app/controllers/api/v1/dependencies_controller.rb#L28

I think something similar to this should work:

```ruby
  def me
    authenticate_or_request_with_http_basic do |username, password|
      if user = AuthenticatedUser.authenticate(username.strip, password)
        respond_to do |format|
          format.json { render json: user }
          format.yaml { render yaml: user }
        end
      else
        respond_to do |format|
          message = { error: "Invalid credentials", code: 401 }
          format.json { render json: message, status: :unauthorized }
          format.yaml { render yaml: message, status: :unauthorized }
        end
      end
    end
  end
```

That way the response is in proper format.

```
curl -H "Authorization: Basic cmV0cm8yOnJldHJvMjIzMw=="  http://localhost:3000/api/v1/profile/me.yaml
---
error: Invalid credentials
code: 401

curl -H "Authorization: Basic cmV0cm8yOnJldHJvMjIzMw=="  http://localhost:3000/api/v1/profile/me.json
{"error":"Invalid credentials","code":401}
```

Currently the response looks like this:

```
curl -H "Authorization: Basic cmV0cm8yOnJldHJvMjIzMw=="  http://localhost:3000/api/v1/profile/me.yaml
HTTP Basic: Access denied.
```

---

_:thinking: Clearly this something more controllers can share (like `Api::V1::ApiKeysController`) and could be refactored to `Api::BaseController` next to those similar methods. But that is probably out of scope for this PR._

https://github.com/rubygems/rubygems.org/blob/1b75f7a8d059042d8f3faa4075f09d3b3988d96e/app/controllers/api/base_controller.rb#L47-L57